src/descriptors/doc.md

```diff
@@ -0,0 +1,82 @@
+# Marina output descriptors template
+The `descriptors` package extends [the original syntax of output script descriptors](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md).
+
+Giving a string template, the `evaluate` function is able to compute:
+* redeem script 
+* witness scripts if needed (the unsigned part of the witness).
+
+## Example
+
+Assuming `xpub0123456789` is one of the Marina accounts master extended public key, and given the following template: 
+
+```
+elp2wsh(asm(xpub0123456789 OP_CHECKSIG))
+```
+
+Marina will creates a `Context` object mapping each `xpub...` in the template to their actual derivation path:
+
+* in case of signing an input: the path equals the one of the utxo.
+* in case of create a new address: the path equals the one of the next address.
+
+Then calling `evaluate` we are able to get our witness script and redeem script:
+
+```ts
+const result = evaluate(marinaContext, "elp2wsh(asm(xpub0123456789 OP_CHECKSIG))")
+const witness = result.witnesses[0] // segwit v0 has 1 witness
+const redeemScript = result.redeemScript // OP_O .... 
+```
+
+> The `result` object value depends on the Context! 
+
+## API
+
+All descriptors can accept `xpub...` token, the public keys are replaced before parsing the template.
+
+### Raw
```

Comment:
`raw` should only be used at top level imho, like original descriptors and...shouldn't be the scriptPubKey? (not redeem script as you mention)